### PR TITLE
Final Additions and Changes for the Json Library

### PR DESCRIPTION
**Pull request by bythos14**
Switched ValueType to an enum class. Changed values to be more C++ like, rather than C like.

Added public type specifier to the InitParameter class.

Removed Initializer::setAllocatorInfoCallBack and AllocatorInfoCallback typedef.

Added some missing documentation, and some other minor changes.

### CODE REVIEW

**Comment by d3m3vilurr** [comment]
please recheck the indentations. they mixed soft-hard tab

**Comment by bythos14** [comment]
Fixed the indentations.

File: include/psp2/json.h [hunk 1]
```diff
@@ -973,7 +1050,7 @@ class Parser
 	 * 					  current character.
 	 * @param[in]  data - Data to pass to the callback function
 	 */
-	static int parse(Value& val, ParseCallback cb, void* data);
+	static int parse(Value& val, ParseCallBack cb, void* data);
```
Comment (d3m3vilurr) [inline include/psp2/json.h:1053]: I preper `Callback` than `CallBack`; now a day, it's an one word.
so how about change the `db.yml`?

Comment (bythos14) [reply include/psp2/json.h]: Made the change in db.yml.

**Comment by d3m3vilurr** [comment]
LGTM :+1: thank you

**Pull request merged**